include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -467,6 +502,15 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
             __kernel,
 #endif
             sycl::nd_range<1>(__segments * __sg_size, __sg_size), [=](sycl::nd_item<1> __self_item) {
+
+                //Optimization: skip re-order phase if the all keys are the same, do just copying
+                if(__no_op_acc[0])
+                {
+                    __copy_kernel_for_radix_sort(__segments, __blocks_per_segment * __block_size, __sg_size, __self_item,
```

Comment:
Have you considered avoiding copying at all? This would require adding some logic in case of odd number of such skipped copies, though.

Reply:
I believe there is no need in barriers, but yes, it will result in extra overheads because of:
- Need to support an additional flag to indicate if the source and destination buffers must be swapped. It will add an extra global memory read and write in `scan` kernel.
- Need to support logic which copies the data into the main buffer if it is the last iteration and the data is in the temporary buffer. This will add 2 extra if conditions into `reorder` kernel (checking the new flag before potential swapping, checking if it is the last stage), swap operation, and an additional read of the new flag.

Anyway, thanks for the investigation. If necessary, it can be further investigated in a separate PR.

Reply:
Actually, we would need a global synchronization if we perform copying at the end of `reorder` kernel when the data is not in the main buffer after the last sort iteration. Thus it must be done in a separate kernel. So yes, it is difficult to implement, I agree with the current approach.